src/renderer/families/polkadot/BondFlowModal/Body.js

```diff
@@ -0,0 +1,197 @@
+import StepConfirmation, { StepConfirmationFooter } from "./steps/StepConfirmation";
+import logger from "~/logger/logger";
+
+type OwnProps = {|
+  stepId: StepId,
+  onClose: () => void,
+  onChangeStepId: StepId => void,
+  params: {
+    account: Account,
+    parentAccount: ?Account,
+    reward: number,
+  },
+  name: string,
+|};
+
+type StateProps = {|
+  t: TFunction,
+  device: ?Device,
+  accounts: Account[],
+  device: ?Device,
+  closeModal: string => void,
+  openModal: string => void,
+|};
+
+type Props = OwnProps & StateProps;
+
+const steps: Array<St> = [
+  {
+    id: "amount",
+    label: <Trans i18nKey="polkadot.bond.steps.amount.title" />,
+    component: StepAmount,
+    noScroll: true,
+    footer: StepAmountFooter,
+  },
+  {
+    id: "connectDevice",
+    label: <Trans i18nKey="polkadot.bond.steps.connectDevice.title" />,
+    component: GenericStepConnectDevice,
+    onBack: ({ transitionTo }: StepProps) => transitionTo("amount"),
+  },
+  {
+    id: "confirmation",
+    label: <Trans i18nKey="polkadot.bond.steps.confirmation.title" />,
+    component: StepConfirmation,
+    footer: StepConfirmationFooter,
+  },
+];
+
+const mapStateToProps = createStructuredSelector({
+  device: getCurrentDevice,
+});
+
+const mapDispatchToProps = {
+  closeModal,
+  openModal,
+};
+
+const Body = ({
+  t,
+  stepId,
+  device,
+  closeModal,
+  openModal,
+  onChangeStepId,
+  params,
+  name,
+}: Props) => {
+  const [optimisticOperation, setOptimisticOperation] = useState(null);
+  const [transactionError, setTransactionError] = useState(null);
+  const [signed, setSigned] = useState(false);
+  const dispatch = useDispatch();
+
+  const {
+    transaction,
+    setTransaction,
+    account,
+    parentAccount,
+    status,
+    bridgeError,
+    bridgePending,
+  } = useBridgeTransaction(() => {
+    const { account, parentAccount } = params;
+
+    const bridge = getAccountBridge(account, parentAccount);
+
+    const t = bridge.createTransaction(account);
+
+    const transaction = bridge.updateTransaction(t, {
+      mode: "bond",
+      recipient: account.freshAddress,
+      rewardDestination: "Stash",
+    });
+
+    return { account, parentAccount, transaction };
+  });
+
+  const handleCloseModal = useCallback(() => {
+    closeModal(name);
+  }, [closeModal, name]);
+
+  const handleStepChange = useCallback(e => onChangeStepId(e.id), [onChangeStepId]);
+
+  const handleRetry = useCallback(() => {
+    onChangeStepId("amount");
+  }, [onChangeStepId]);
+
+  const handleTransactionError = useCallback((error: Error) => {
+    if (!(error instanceof UserRefusedOnDevice)) {
+      logger.critical(error);
+    }
+    setTransactionError(error);
+  }, []);
+
+  const handleOperationBroadcasted = useCallback(
+    (optimisticOperation: Operation) => {
+      if (!account) return;
+      dispatch(
+        updateAccountWithUpdater(account.id, account =>
+          addPendingOperation(account, optimisticOperation),
```

Comment:
it's a copy/paste from cosmos or tron.
I try not to alter things already done on other coins, and avoid adding entropy in the technical debt, so that if a refacto/maintance is done, it's easier to apply it on multiple non-DRY code.